ledger/ledger-on-sql/src/app/scala/com/daml/ledger/on/sql/SqlLedgerFactory.scala

```diff
@@ -18,16 +19,19 @@ import com.daml.ledger.participant.state.kvutils.caching._
 import com.daml.ledger.resources.{Resource, ResourceContext, ResourceOwner}
 import com.daml.lf.engine.Engine
 import com.daml.logging.LoggingContext
-import com.daml.platform.configuration.LedgerConfiguration
+import com.daml.platform.configuration.InitialLedgerConfiguration
 import scopt.OptionParser
 
 object SqlLedgerFactory extends LedgerFactory[ReadWriteService, ExtraConfig] {
   override val defaultExtraConfig: ExtraConfig = ExtraConfig(
     jdbcUrl = None
   )
 
-  override def ledgerConfig(config: Config[ExtraConfig]): LedgerConfiguration =
-    super.ledgerConfig(config).copy(initialConfigurationSubmitDelay = Duration.ZERO)
+  override def initialLedgerConfig(config: Config[ExtraConfig]): InitialLedgerConfiguration =
+    InitialLedgerConfiguration(
+      configuration = Configuration.reasonableInitialConfiguration,
+      delayBeforeSubmitting = Duration.ZERO,
```

Comment:
Doesn't a delay of zero mean the participant will submit a new config change request every time it restarts? Even with an up-to-date index database, it will take some time until `CurrentLedgerConfiguration` sees a configuration (because of the round trip time to the database).

Reply:
Yup, you are absolutely right. I screwed this up thinking this was Daml-on-SQL code, not ledger-on-sql.